### ISSUE TITLE
[Feature request] Rule no-undefined-styles

### ISSUE THREAD

**Issue by ncuillery** [Intellicode/eslint-plugin-react-native#144]
After a refactoring, I left my code with undefined entries in the StyleSheet. No errors at runtime but it has lead to cosmetic bugs, hard to catch.

It would have been great if eslint had warned me.

Example of invalid code: (`styles.startButton` and `styles.disabled` aren't defined)
```javascript
const styles = StyleSheet.create({
  startView: {
    justifyContent: 'center',
  },
});

render() {
  return (
    <View style={styles.startView}>
      <Button style={[styles.startButton, this.props.disabled && styles.disabled]} />
    </View>
  );
}
```

This rule would be kind of the opposite of `no-unused-styles`.

**Comment by Intellicode** [comment]
It's on my list for quite a while, if you want to make a PR to get things going I'm more than happy to integrate it in the library. For time being lets keep track of this feature in the previous issue: https://github.com/Intellicode/eslint-plugin-react-native/issues/14
